modules/openapi-generator/src/test/java/org/openapitools/codegen/TestUtils.java

```diff
@@ -81,4 +83,19 @@ public static void ensureDoesNotContainsFile(Map<String, String> generatedFiles,
         }
         assertFalse(generatedFiles.containsKey(absoluteFilename), "File '" + absoluteFilename + "' was found in the list of generated files");
     }
+
+    public static void validateJavaSourceFiles(Map<String, String> fileMap) {
+        fileMap.forEach( (fileName, fileContents) -> {
+                if (fileName.endsWith(".java")) {
+                    assertValidJavaSourceCode(fileContents);
+                }
+            }
+        );
+    }
+
+    public static void assertValidJavaSourceCode(String javaSourceCode) {
+        // note: "StaticJavaParser.parse" throws an exception if the parser
+        // detects a syntax error
+        assertTrue(StaticJavaParser.parse(javaSourceCode).getTypes().size() > 0);
```

Comment:
Could the error message be improved when the assertion is failing? 
Maybe if you pass the `fileName` in `validateJavaSourceFiles(..)` you could add this to the error message printed when the condition is `false`. This will help to understand where to look to reproduce the issue.